vowpalwabbit/bfgs.cc

```diff
@@ -1090,24 +1090,18 @@ base_learner* bfgs_setup(options_i& options, vw& all)
   bool bfgs_option = false;
   option_group_definition bfgs_outer_options("LBFGS and Conjugate Gradient options");
   bfgs_outer_options.add(
-      make_option("conjugate_gradient", conjugate_gradient).keep().help("use conjugate gradient based optimization"));
+      make_option("conjugate_gradient", conjugate_gradient).keep().necessary().help("use conjugate gradient based optimization"));
```

Comment:
this reduction is created when `conjugate_gradient || bfgs`. Only one is required, not both. Is there a way to differentiate that here? Maybe that gets handled by the outer/inner_options, it just seems a little confusing

Reply:
they are two separate option_group_definition. It is differentiated in that way, by being tracked by different instantiations. The new if statement is symmetrical to the past if statement, let me know if you have a suggestion to make it less confusing? It seemed the same as the previous impl to me.